src/esp/gfx/CubeMap.cpp

```diff
@@ -0,0 +1,507 @@
+        coordStrings[iFace], imageFileExtension);
+
+    importer->openFile(filename);
+    Cr::Containers::Optional<Mn::Trade::ImageData2D> imageData =
+        importer->image2D(0);
+
+    // XXX
+    /*
+    Mn::Debug{} << "imagedata raw data size = " << (*imageData).data().size();
+    {
+      auto view = Cr::Containers::arrayCast<const float>((*imageData).data());
+      Mn::Debug{} << "first 4 rgbs from imagedata"
+                  << view.prefix(12);  // first 12 floats --> 4 Rgbs
+    }
+    */
+
+    // sanity checks
+    CORRADE_INTERNAL_ASSERT(imageData);
+    Mn::Vector2i size = imageData->size();
+    CORRADE_ASSERT(
+        size.x() == size.y(),
+        " CubeMap::loadTexture(): each texture image must be a square.", );
+    if (iFace == 0) {
+      imageSize = size.x();
+      reset(imageSize);
+    } else {
+      CORRADE_ASSERT(size.x() == imageSize,
+                     " CubeMap::loadTexture(): texture images must have the "
+                     "same size.", );
+    }
+
+    switch (type) {
+      case TextureType::Color:
+        texture->setSubImage(convertFaceIndexToCubeMapCoordinate(iFace), 0, {},
+                             *imageData);
+        break;
+
+      case TextureType::Depth: {
+        // The pixel format for depth texture is R32F. When it is saved as hdr,
+        // the single channel is expanded to three channels by repeating the R
+        // channel 3 times (and it becomes RGB32F).
+        // That means when it is loaded, we need to dedup by taking just the
+        // first component (R component) out of each pixel.
+        Cr::Containers::StridedArrayView2D<const Mn::Color3>
+            imageStridedArrayView = imageData->pixels<Mn::Color3>();
+
+        //  This casts the Color3 view to a float, which effectively uses only
+        //  four bytes of each element, which is the red channel
+        Cr::Containers::StridedArrayView2D<const float> red =
+            Cr::Containers::arrayCast<const float>(imageStridedArrayView);
+
+        Cr::Containers::Array<float> depthImage{
+            Cr::Containers::NoInit, static_cast<size_t>(size.product())};
+        // Turns the `depthImage` into a tightly packed 2D view of the same rows
+        // and pixels as `imageStridedArrayView`
+        Cr::Containers::StridedArrayView2D<float> output{
+            depthImage, {std::size_t(size.y()), std::size_t(size.x())}};
+
+        // copy the data
+        Cr::Utility::copy(red, output);
+
+        // XXX
+        /*
+        unsigned int idx = 0;
+        for (auto row : imageStridedArrayView) {
+          for (const Mn::Color3& pixel : row) {
+            depthImage[idx++] = pixel.r();
+          }
+        }
+        */
+        // Mn::Debug{} << "depthImage (top 9) = " << depthImage.prefix(9);
+        // Mn::Debug{} << "imageData size: " << imageData->size();
+        Mn::ImageView2D imageView(Mn::GL::PixelFormat::DepthComponent,
```

Comment:
Note: compared to the previous commit, I replace the R32F by `Mn::GL::PixelFormat::DepthComponent, Mn::GL::PixelType::Float`

Hope it is the correct thing to do.